src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/Transition.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.models
+
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.CronSchedule
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.ScheduleParser
+import org.elasticsearch.ElasticsearchParseException
+import org.elasticsearch.common.unit.ByteSizeValue
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.ToXContentObject
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+
+data class Transition(
+    val stateName: String,
+    val conditions: Conditions?
+) : ToXContentObject {
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        builder.startObject()
+            .field(STATE_FIELD, stateName)
+        if (conditions != null) builder.field(CONDITIONS_FIELD, conditions)
+        return builder.endObject()
+    }
+
+    companion object {
+        const val STATE_FIELD = "state_name"
+        const val CONDITIONS_FIELD = "conditions"
+
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser): Transition {
+            lateinit var name: String
```

Comment:
Let's switch from lateinit usage here to just var name: String? = null
From testing it turns out if the name is missing then when doing requireNotNull(name) it'll throw an UninitializedPropertyAccessException without the message we have which seems like worse behavior.

Reply:
Removed lateinit from all model ```parse``` methods.